feat: stakefish integration

zhaono1 opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Full Cardano (ADA) staking support with validator selection UI, Stakefish provider flows, and new staking dialogs/components
  * Partial withdrawals and protocol-provided decimals override for amount inputs
  * Hoodi testnet added

* **Bug Fixes & Improvements**
  * Improved staking signing/withdraw flows (including Stakefish message/signature handling), enhanced portfolio testnet handling, and UI refinements (disabled items, badges, footers)

* **Chores**
  * Release version bumped; spellchecker skip words updated

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

zhaono1 committed:
feat: integrate Stakefish staking support for ADA and ETH

zhaono1 committed:
feat: add ADA management components and enhance staking actions

zhaono1 committed:
feat: add ADA management components and enhance staking actions

zhaono1 committed:
Merge branch 'x' into feat/stakefish-integration-v2

zhaono1 committed:
feat: integrate universal withdraw functionality and enhance risk notice handling in AdaManageContent

wabicai committed:
fix: snap linux running error  (#9105)

* Revert "fix: Add raw-usb plugs in linux snap for web-usb connect (#8887)"

This reverts commit e3646d4c28117c8bfcc6235f233d575e4fa0c5ba.

* fix: the lack snap plugs

* fix: replace with default

weatherstar committed:
optimize: recent network OK-46450 (#9107)

* chore: update locale

* feat: pin recent networks in account chain selector

* feat: pin recent networks in pure chain selector

ByteZhang1024 committed:
feat: support bitcoin only firmware (#8948)

* feat: support bitcoin only firmware

* feat: fix bitcoin only ui

sidmorizon committed:
Fix/update dev settings (#9108)

* fix: update dev settings

* fix: lint

* fix: lint

franco-chan committed:
change android splash screen logo (#9112)

limichange committed:
Perps reward dashboard (#9098)

* feat: enhance OnChainReward component to support perpetual rewards

- Introduced usePerpReward hook to manage perpetual rewards.
- Updated OnChainReward component to display both on-chain and perpetual rewards.
- Refactored reward summary calculations for improved clarity and efficiency.
- Modified types to accommodate new reward structures.

* feat: enhance OnChainReward component to support perpetual rewards

- Introduced usePerpReward hook to manage perpetual rewards.
- Updated OnChainReward component to display both on-chain and perpetual rewards.
- Refactored reward summary calculations for improved clarity and efficiency.
- Modified types to accommodate new reward structures.

* feat: add support for reward tab selection in referral components

- Introduced EExportTab enum to manage reward tab states.
- Updated ExportButton to accept tab as a prop for export functionality.
- Enhanced EarnRewardPageWrapper to manage active reward tab state.
- Modified RewardTypeTabs to handle tab changes and initial tab selection.
- Adjusted TradingVolumeSummaryCard and RecordsAccordionList for UI consistency.
- Updated ServiceReferralCode to include tab in export parameters.

* feat: enhance EarnReward page with new accordion components

- Added ReferFriendsAccordionItem and ReferFriendsListHeader components for improved UI structure.
- Refactored RewardAccountList and RecordsAccordionList to utilize new accordion components for better code organization.
- Adjusted EarnRewardPageWrapper layout for responsive design.
- Updated component exports to include new items for better accessibility.

* refactor: simplify tab selection logic in PerpsRecordsTab component

- Streamlined the useEffect hook to manage active tab selection based on available and total data.
- Removed redundant conditions for setting the active tab, improving code clarity and maintainability.
- Adjusted layout structure by removing unnecessary Divider component for a cleaner UI.

* update

* update

franco-chan committed:
find-turning onboarding background color (#9111)

Minnzen committed:
refactor: enhance perps trading with rebate binding and UI improvements (#9114)

* fix: adjust font scaling in TradingFormInput component

* refactor: streamline trading view message handler and enhance trading panel components

- Simplified the trading view message handler by removing unnecessary date calculations and directly loading trade history.
- Introduced debouncing for margin required and liquidation price calculations in the trading panel for improved performance.
- Updated the trading form to calculate available to trade based on maximum values from active asset data.

* refactor: add internal rebate binding check to ServiceHyperliquid

- Introduced a new method to check internal rebate binding status for accounts.
- Updated account status details to include internal rebate binding validation.
- Implemented non-blocking checks for internal rebate binding during account activation process.

Minnzen committed:
refactor: enhance TradesHistoryRow component to include liquidation details (#9115)

- Updated the TradesHistoryRow component to display liquidation information in trade direction.
- Added logic to differentiate between market and backstop liquidations.

Minnzen committed:
refactor: streamline ServiceHyperliquid methods and improve rebate binding logic (#9116)

- Updated methods in ServiceHyperliquid to replace reversed parameter with aggregateByTime for better data handling.
- Refactored rebate binding logic to enhance clarity and maintainability, including renaming methods for better understanding.
- Improved error handling and streamlined checks for internal rebate binding status during account operations.

ByteZhang1024 committed:
fix: multi device connection check for updates (#9113)

limichange committed:
fix: restore 60s auto-refresh for market v2 token list (#9110)

- Add pollingInterval back to useMarketTokenList hook
- This was accidentally removed in PR #8321

weatherstar committed:
fix: recent networks flex error (#9117)

limichange committed:
feat: enhance referral features and navigation updates (#9109)

* feat: enhance referral features and navigation updates

- Refactor deep link handling to switch to tab navigation for referral actions.
- Introduce a new referral landing page with app redirect support.
- Add a new component for displaying referral benefits.
- Update routing to include a new InvitedByFriend page.
- Improve referral code handling in wallet binding logic.
- Adjust layout and styling in referral components for better responsiveness.

* useGetReferralCodeWalletInfo

* feat: implement invited by friend deep link handling

- Add support for deep linking via the invited_by_friend path.
- Enhance navigation to switch tabs based on the page parameter.
- Update InvitedByFriend component to accept and utilize the page parameter.
- Modify referral actions to include page context for improved user experience.

* refactor: update import paths for ReferralBenefitsList component

- Changed import paths for ReferralBenefitsList in ReferAFriendHowToPhase and ReferAFriendIntroPhase components to ensure consistency and maintainability.
- Removed relative imports in favor of absolute paths for better clarity.

originalix committed:
feat: coin control (#9118)

* feat: add CoinControlBadge component and integrate into SendDataInputContainer

* feat: implement CoinControl feature with UTXO management and routing integration

* feat: add CoinControl functionality with UTXO retrieval and integration into SendDataInputContainer

* feat: adjust width of UTXOListItem checkbox container for better responsiveness

* feat: enhance Checkbox component with shouldStopPropagation prop and improve sorting logic using BigNumber

* feat: add selected UTXOs management with new atom and actions for Coin Control

* feat: integrate Coin Control feature into SendDataInputContainer with UTXO selection management

* feat: implement Coin Control UTXO filtering and clear selection after transaction

* feat: update coinselect dependency to version 3.1.16 and add skipUtxoSelection option in VaultBtc

* feat: add displayCoinControlButton to manage Coin Control visibility in SendDataInputContainer

* feat: add header right filter button for sorting options in CoinControlPage

* feat: add block time handling for UTXOs in Coin Control and related types

* chore: i18n

* feat: implement logging for selected UTXOs in Coin Control

* feat: add logging for coin control results in SendScene and VaultBtc

* feat: remove unused isBtcLikeNetwork memoization in SendDataInputContainer

ByteZhang1024 committed:
Fix/firmware check OK-45850 OK-46577 (#9119)

* fix: btc only feature

* feat: tron sign message v2 on touch

Minnzen committed:
refactor: improve rebate binding logic and add credential management in ServiceHyperliquid (#9120)

- Enhanced the rebate binding process by introducing a caching mechanism for internal rebate status checks.
- Added a new method to clear local agent credentials, ensuring proper management of user data during rebate binding.
- Updated status details to reflect the outcome of rebate binding checks more accurately.
- Adjusted the reference network ID for consistency in network handling.

revan-zhang commented:
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scanner | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-critical.svg) Critical | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-high.svg) High | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-medium.svg) Medium | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-low.svg) Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/rayston92/pr-checks/c4546133-d1ed-4c4e-be3f-dd28734a34ad?refs=56236d134c0b4273db7fd66ca3966be900881a6c&source=prChecksComment) |
| :white_check_mark: | Licenses| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/rayston92/pr-checks/c4546133-d1ed-4c4e-be3f-dd28734a34ad/license?refs=56236d134c0b4273db7fd66ca3966be900881a6c&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

zhaono1 committed:
fix: earn redesign OK-44365 OK-46583 OK-46555 OK-46534 OK-46530 OK-46529 OK-46480 OK-46359  (#9122)

* fix: refactor staking hooks and remove unused useEarnAccount

* chore: fetch i18n

* fix: add accountId to manage page request parameters

* chore: update i18n

* fix: remove display property from Empty component in EarnBlockedOverview

* fix: update border color in BannerItemV2 component

* fix: disable USDf staking in stakingConfig

* fix: update navigation logic in Earn components to use EarnNavigation utility

originalix committed:
Revert "feat: coin control (#9118)" (#9124)

This reverts commit cc1c7d401ef6ea961c97099a09d005675b843d6d.

originalix committed:
feat: add connection info retrieval and IP connection check (#9129)

* feat: add connection info retrieval and IP connection check in ServiceIpTable

* chore: i18n

* fix: add IP connection check in getBlockRegion method

Minnzen committed:
feat: add configVersion handling and credential management in ServiceHyperliquid (#9127)

- Introduced a new method to remove all HyperLiquid agent credentials when the configVersion changes.
- Updated ISimpleDbPerpData interface to include configVersion.
- Enhanced ServiceHyperliquid to check for configVersion changes and manage agent credentials accordingly.
- Added accountId and accountName parameters to notification service for better tracking.

zhaono1 committed:
Merge branch 'x' into feat/stakefish-integration-v2

originalix committed:
Merge branch 'x' of github.com:OneKeyHQ/app-monorepo into feat/stakefish-integration-v2

originalix committed:
feat: enhance Vault class with internal DApp transaction encoding and update useProtocolDetailBreadcrumb imports

originalix committed:
feat: add hoodi testnet

originalix committed:
Merge branch 'x' of github.com:OneKeyHQ/app-monorepo into feat/stakefish-integration-v2

originalix committed:
fix: update @onekeyfe/cardano-coin-selection-asmjs to version 1.1.8

originalix committed:
Merge branch 'x' of github.com:OneKeyHQ/app-monorepo into feat/stakefish-integration-v2

# Conflicts:
#	packages/kit/src/views/Staking/pages/ManagePosition/components/ManagePositionContent.tsx
#	packages/kit/src/views/Staking/pages/ManagePosition/components/USDEManageContent.tsx

originalix committed:
feat: update Cardano coin selection library to version 1.1.10-alpha.6 and enhance transaction encoding with staking support

originalix committed:
fix: update native amount calculation logic

originalix committed:
fix: update @onekeyfe/cardano-coin-selection-asmjs to version 1.1.10-alpha.7

originalix committed:
feat: enhance KYC dialog with confirmation handling and integrate into AdaManageContent

originalix committed:
feat: add loading state to buttons and enhance dialog handling in staking components

originalix committed:
feat: integrate EarnValidatorSelect component and update related types for validator selection in staking

originalix committed:
feat: add containerStyle prop to BasicTradeOrBuy and UniversalStake components for improved styling flexibility

originalix committed:
feat: remove description from EarnValidatorSelect component for cleaner UI

originalix committed:
feat: integrate Stakefish validator support and enhance permit signature handling in staking components

zhaono1 committed:
Merge branch 'x' into feat/stakefish-integration-v2

zhaono1 committed:
Merge branch 'x' into feat/stakefish-integration-v2

originalix committed:
feat: add identity handling for Stakefish validator and update checkAmount action type

originalix committed:
feat: add support for partial withdrawals in staking configuration and related components

originalix committed:
feat: add signature handling for Stakefish ETH withdrawals and updates to related components

originalix committed:
feat: enhance ServiceStaking and useEarnPortfolio to support testnet networks and improve network ID handling

originalix committed:
Merge branch 'x' of github.com:OneKeyHQ/app-monorepo into feat/stakefish-integration-v2

originalix committed:
chore: bump version to 5.19.0

originalix committed:
feat: add optional description field to EarnSelectOption and update EarnValidatorSelect to utilize it

originalix committed:
Merge branch 'x' of github.com:OneKeyHQ/app-monorepo into feat/stakefish-integration-v2

originalix committed:
Merge branch 'x' of github.com:OneKeyHQ/app-monorepo into feat/stakefish-integration-v2

originalix committed:
feat: add protocolInputDecimals to manage position and staking components for improved decimal handling

originalix committed:
feat: update decimal handling in Claim, Withdraw, and WithdrawSection components to use protocolInputDecimals

originalix committed:
Merge branch 'x' of github.com:OneKeyHQ/app-monorepo into feat/stakefish-integration-v2

originalix committed:
feat: enhance OptionList and WithdrawOptions components with support for disabled items and descriptions

originalix committed:
Merge branch 'x' of github.com:OneKeyHQ/app-monorepo into feat/stakefish-integration-v2

originalix committed:
Merge branch 'x' of github.com:OneKeyHQ/app-monorepo into feat/stakefish-integration-v2

originalix committed:
feat: add poolKeyHash to IAdaStakingCertificate and update address handling for staking transactions in Vault

originalix committed:
fix: update @onekeyfe/cardano-coin-selection-asmjs to version 1.1.10-alpha.8 and adjust transaction signing logic

originalix committed:
Merge branch 'x' of github.com:OneKeyHQ/app-monorepo into feat/stakefish-integration-v2

originalix committed:
feat: enhance staking transaction handling with improved certificate type management and deposit logic

originalix committed:
feat: enhance staking transaction handling with improved staking info management and UI rendering

originalix commented:
@coderabbitai resolve

originalix committed:
Merge branch 'x' into feat/stakefish-integration-v2

ByteZhang1024 commented on packages/kit-bg/src/vaults/impls/ada/KeyringHardware.ts:
```diff
@@ -254,6 +254,11 @@ export class KeyringHardware extends KeyringHardwareBase {
+      '=====>>hw cardanoParams: ',
+      JSON.stringify(cardanoParams, null, 2),
+    );
+
```
remove

originalix replied:
Fixed，接入日志了

limichange committed:
Merge branch 'x' into feat/stakefish-integration-v2

originalix committed:
feat: enhance ADA transaction logging and add new spell checker skip word

originalix committed:
Merge branch 'x' of github.com:OneKeyHQ/app-monorepo into feat/stakefish-integration-v2

# Conflicts:
#	packages/core/package.json
#	yarn.lock

originalix committed:
Merge branch 'x' into feat/stakefish-integration-v2
